### PR TITLE
tree.rs: lift id and name into a struct type

### PR DESCRIPTION
**Pull request by hlian**
use a new variant type `Contents` to disambiguate leaves from nodes

### CODE REVIEW

**Comment by hlian** [comment]
my first github pr in four years

File: src/tree.rs [hunk 1]
```diff
@@ -1,46 +1,24 @@
...
-        match self {
-            Tree::Leaf { name, .. } => name,
-            Tree::Node { name, .. } => name,
-        }
-    }
+#[derive(Debug)]
+pub struct Tree {
+    pub(crate) id: u64,
+    pub(crate) name: String,
+    pub(crate) contents: Contents,
```
Comment (soenkehahn) [inline src/tree.rs]: Currently this crate doesn't have a library, so `pub` is the same as `pub(crate)`. That's why I made everything `pub` that needs is used in another module. Maybe do that too here? TIOLI!

```suggestion
    pub id: u64,
    pub name: String,
    pub contents: Contents,
```

File: src/fuse.rs [hunk 2]
```diff
@@ -78,13 +79,16 @@ impl Filesystem for TreeFs {
...
+                    ..
+                }) => {
                     for child in children {
                         assert!(!reply.add(
-                            child.id(),
+                            child.id,
                             1,
                             TreeFs::tree_to_file_type(child),
-                            child.name()
+                            child.name.to_owned()
```
Comment (soenkehahn) [inline src/fuse.rs]:
```suggestion
                            &child.name
```
`add` is polymorphic in this argument and is fine with `&str`.

**Review by soenkehahn — APPROVED** [review]
Very nice!

**Pull request merged**